src/main/java/com/google/api/codegen/transformer/ApiMethodTransformer.java

```diff
@@ -277,15 +280,17 @@ private void setListMethodFields(
               typeTable,
               resourceField,
               resourceField.getType().makeOptional(),
-              resourceFieldConfig.getEntityName());
+              resourceFieldConfig.getEntityName(),
+              resourceNameType);
     } else {
       resourceTypeName = typeTable.getAndSaveNicknameForElementType(resourceField.getType());
     }
 
     String iterateMethodName =
         context
             .getNamer()
-            .getPagedResponseIterateMethod(context.getFeatureConfig(), resourceFieldConfig);
+            .getPagedResponseIterateMethod(
+                context.getFeatureConfig(), resourceFieldConfig, resourceNameType);
```

Comment:
Yes, that does make a lot of sense. The reason I haven't done that is related to the design choice between having the Config classes reflect the config proto or not.

So having the resourceNameType be a property of the field config would be more require more complex construction and be less reflective of the proto, but would reduce the complexity of the transformers.

If we are happy to break the connection between config classes and config proto, I think this is a sensible change.